kaska/s2_observations.py

```diff
@@ -219,133 +263,181 @@ def read_time_series(self, time_grid):
-        # So valid pixels if all refl > 0 AND mask is True
+
+        # Ensure all surface reflectance pixels have values above 0 and
+        # aren't cloudy. So pixels are valid if all refl > 0 AND mask is True
         # Array of the desired bands. Not necessarily contiguous.
         sel_bands = np.array([1, 2, 3, 4, 5, 6, 7, 8])
         mask1 = np.logical_and(
             np.all(rho_surface[sel_bands] > 0, axis=0), mask
         )
         mask = mask1
         if mask.sum() == 0:
-            LOG.info("%s -> No clear observations", str(timestep))
+            LOG.info(f"{timestep} -> No clear observations")
+            #print(f"{timestep} -> No clear observations")
             return None, None, None, None, None, None
         LOG.info(
-            f"{str(timestep):s} -> Total of {mask.sum():d} clear pixels "
+            f"{timestep} -> Total of {mask.sum():d} clear pixels "
             + f"({100.*mask.sum()/np.prod(mask.shape):f}%)"
         )
 
+        # Rescaling and setting missing pixels to NaN
+        # reflectivity
         rho_surface = rho_surface / 10000.0
-
+        rho_surface[:, ~mask] = np.nan
+        # uncertainty
         rho_unc = np.array(rho_unc) / 10000.0
         rho_unc[:, ~mask] = np.nan
         # Average uncertainty over the image
         rho_unc = np.nanmean(rho_unc, axis=(1, 2))
-        # Set missing pixels to NaN
-        rho_surface[:, ~mask] = np.nan
-        # Now read angles
+
+        # Read in angles
+        #print(next(current_folder.glob("**/SAA_SZA.tif")))
+        try:
+            angle_file_S = next(current_folder.glob("**/SAA_SZA.tif"))
+            #print(angle_file_S)
+        except StopIteration:
+            LOG.info(f"Sun angle file SAA_SZA.tif does not exist.")
+            return None, None, None, None, None, None
+        try:
+            angle_file_V = next(current_folder.glob("**/VAA_VZA_B05.tif"))
+        except StopIteration:
+            LOG.info(f"View angle file VAA_VZA_B05.tif does not exist.")
+            return None, None, None, None, None, None
+
         sun_angles = reproject_data(
-            str(current_folder.parent / "ANG_DATA/SAA_SZA.tif"),
+            str(angle_file_S),
             target_img=self.state_mask,
             x_res=20,
             y_res=20,
             resample=0,
         ).ReadAsArray()
         view_angles = reproject_data(
-            str(current_folder.parent / "ANG_DATA/VAA_VZA_B05.tif"),
+            str(angle_file_V),
             target_img=self.state_mask,
             x_res=20,
             y_res=20,
             resample=0,
         ).ReadAsArray()
+        # zenith angles: 90 - Altitude
         sza = np.cos(np.deg2rad(sun_angles[1].mean() / 100.0))
         vza = np.cos(np.deg2rad(view_angles[1].mean() / 100.0))
+        # azimuth
         saa = sun_angles[0].mean() / 100.0
         vaa = view_angles[0].mean() / 100.0
         raa = np.cos(np.deg2rad(vaa - saa))
         return rho_surface, mask, sza, vza, raa, rho_unc
```

Comment:
The above (from l361) can be DRYer with a function:
```python
def reproject_angles(file_type):
    try:
        angle_file = next(current_folder.glob(f"**/{file_type}.tif"))
    except StopIteration:
        raise ValueError(f"file {file_type}.tif does not exist.")
   angles = reproject_data(
            str(current_folder.parent / f"ANG_DATA/{file_type}.tif"),
            str(angle_file),
            target_img=self.state_mask,
            x_res=20,
            y_res=20,
            resample=0,
        ).ReadAsArray()
    # zenith (90 - altitude)
    za = np.cos(np.deg2rad(angles[1].mean() / 100.0))
    # azimuth
    aa = sun_angles[0].mean() / 100.0
    return za, aa

try:
    sun_angles = reproject_angles('SAA_SZA')
    view_angles = reproject_angles('VAA_VZA_B05')
except ValueError as err:
    LOG.info(f"Sun or View angle {err}")
    return (None,) *  6

raa = np.cos(np.deg2rad(view_angles[1] - sun_angles[1]))
return rho_surface, mask, sun_angle[0], view_angle[0], raa, rho_unc
```

Reply:
Fiiiiiineeeeee.....